src/cache.cpp

```diff
@@ -43,7 +47,7 @@ CacheTree::CacheTree(size_t nsamples, size_t nrules, double c, rule_t *rules,
 }
 
 CacheTree::~CacheTree() {
-    if (root_)
```

Comment:
We originally had this in here because it was segfaulting, so I'd want to make sure we test this behavior carefully

Reply:
Yeah, I tested it very carefully. As you had it, it worked fine with GMP, but without GMP for some reason it would segfault when the optimal rulelist was just the default prediction. Replacing '_root' with 'num_nodes()' fixed that, and now it doesn't segfault with GMP or without GMP.